src/main/java/elements/BaseElementFacadeImpl.java

```diff
@@ -0,0 +1,213 @@
+package elements;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebDriverException;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.pagefactory.ElementLocator;
+import org.openqa.selenium.support.ui.ExpectedCondition;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import net.jodah.failsafe.Failsafe;
+import net.serenitybdd.core.annotations.findby.By;
+import net.serenitybdd.core.pages.DefaultTimeouts;
+import net.serenitybdd.core.pages.WebElementFacade;
+import net.serenitybdd.core.pages.WebElementFacadeImpl;
+import net.serenitybdd.core.selectors.Selectors;
+import net.thucydides.core.ThucydidesSystemProperty;
+import net.thucydides.core.guice.Injectors;
+import net.thucydides.core.util.EnvironmentVariables;
+import net.thucydides.core.webdriver.javascript.JavascriptExecutorFacade;
+import utils.ExceptionLauncher;
+import utils.RetryOnFailure;
+
+public class BaseElementFacadeImpl extends WebElementFacadeImpl implements BaseElementFacade {
+
+	static final Logger LOGGER = LogManager.getLogger();
+	ExceptionLauncher exceptionLauncher;
+
+	protected WebDriver driver;
+	protected WebElement webElement;
+
+	public BaseElementFacadeImpl(WebDriver driver, ElementLocator locator, WebElement element,
+			long implicitTimeoutInMilliseconds, long waitForTimeoutInMilliseconds) {
+		super(driver, locator, element, implicitTimeoutInMilliseconds, waitForTimeoutInMilliseconds);
+		this.driver = driver;
+		exceptionLauncher = new ExceptionLauncher();
+	}
+
+	/**
+	 * This method has the goal of creating a BaseElementFacade instance from a
+	 * WebElementFacade instance. Consequently, it would have the access to the
+	 * BaseElementFacade interface methods
+	 * 
+	 * @return a wrapped BaseElementFacade of the webElementFacade
+	 */
+	@SuppressWarnings("unchecked")
+	public static <T extends BaseElementFacade> T wrapWebElementFacade(final WebDriver driver,
+			final WebElementFacade element, final long implicitTimeoutInMilliseconds,
+			final long waitForTimeoutInMilliseconds) {
+		return (T) new BaseElementFacadeImpl(driver, null, element, implicitTimeoutInMilliseconds,
```

Comment:
```suggestion
		return (T) new BaseElementFacadeImpl(getDriver(), null, element, implicitTimeoutInMilliseconds,
```
We have already access to WebDriver IMO.